src/main/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920Codec.java

```diff
@@ -5,28 +5,42 @@
 package org.opensearch.knn.index.codec.KNN920Codec;
 
 import lombok.Builder;
+import lombok.extern.log4j.Log4j2;
 import org.apache.lucene.codecs.Codec;
 import org.apache.lucene.codecs.CompoundFormat;
 import org.apache.lucene.codecs.DocValuesFormat;
 import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.lucene92.Lucene92HnswVectorsFormat;
+import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.knn.common.KNNConstants;
 import org.opensearch.knn.index.codec.KNNFormatFacade;
 import org.opensearch.knn.index.codec.KNNFormatFactory;
+import org.opensearch.knn.index.mapper.KNNVectorFieldMapper;
+
+import java.util.Map;
+import java.util.Optional;
 
 import static org.opensearch.knn.index.codec.KNNCodecFactory.CodecDelegateFactory.createKNN92DefaultDelegate;
 
 /**
  * KNN codec that is based on Lucene92 codec
  */
+@Log4j2
 public final class KNN920Codec extends FilterCodec {
 
     private static final String KNN920 = "KNN920Codec";
+
     private final KNNFormatFacade knnFormatFacade;
+    private final Optional<MapperService> mapperService;
```

Comment:
Why do we need optional member?

Reply:
it can be null if codec initialized through the SPI, it's a valid case. In core they check it for null explicitly, in our code I'd like to highlight that it can be null explicitly.

Reply:
Got it. Thanks for clarification.